src/Router.sol

```diff
@@ -29,6 +31,9 @@ contract Router is IRouter, EIP712, FeeGenerator {
     /// @notice Mapping for recording exclusive agent contract to each user
     mapping(address user => IAgent agent) public agents;
 
+    /// @notice Mapping for user of each delegatee
+    mapping(address delegatee => address user) public delegators;
+
```

Comment:
Does this mapping indicate that one delegatee can only serve one user? It seems that other users can point the delegatee to other users through allow.

Reply:
Yes I suppose that should be the case. In this scenario, the delegatee will be the one to call router.
- If `user` be the key -> `user` can only have one `delegatee`, `delegatee` can serve multiple `users` -> `delegatee` owns multiple authorities which might need addithional permission design  
- If `delegatee` be the key -> `delegatee` can only serve one `user`, `user` can have multiple delegatee -> the `delegatee` execution only relates to a single `user`